zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/utils/CamundaExporterITInvocationProvider.java

```diff
@@ -0,0 +1,161 @@
+import org.testcontainers.elasticsearch.ElasticsearchContainer;
+
+public class CamundaExporterITInvocationProvider
+    implements TestTemplateInvocationContextProvider, AfterAllCallback, BeforeAllCallback {
+
+  public static final String CONFIG_PREFIX = "camunda-record";
+  private final ElasticsearchContainer elsContainer =
+      TestSupport.createDefeaultElasticsearchContainer();
+  private final OpensearchContainer<?> osContainer = TestSupport.createDefaultOpensearchContainer();
+
+  private final List<AutoCloseable> closeables = new ArrayList<>();
+
+  private ExporterConfiguration getConfigWithConnectionDetails(
+      final ConnectionTypes connectionType) {
+    final var config = new ExporterConfiguration();
+    config.getIndex().setPrefix(CONFIG_PREFIX);
+    config.getBulk().setSize(1); // force flushing on the first record
+    switch (connectionType) {
+      case ELASTICSEARCH -> config.getConnect().setUrl(elsContainer.getHttpHostAddress());
+      case OPENSEARCH -> config.getConnect().setUrl(osContainer.getHttpHostAddress());
+      default -> throw new IllegalArgumentException("Unknown connection type: " + connectionType);
+    }
+    config.getConnect().setType(connectionType.getType());
+    return config;
+  }
+
+  @Override
+  public void beforeAll(final ExtensionContext context) {
+    elsContainer.start();
+    osContainer.start();
+
+    closeables.add(elsContainer);
+    closeables.add(osContainer);
+  }
+
+  @Override
+  public boolean supportsTestTemplate(final ExtensionContext extensionContext) {
+    // we only want to template tests in the class which ask for configuration and the client
+    // adapter.
+    final var testParams = extensionContext.getTestMethod().orElseThrow().getParameters();
+    if (testParams.length != 2) {
+      return false;
+    }
+
+    return (testParams[0].getType().equals(ExporterConfiguration.class)
+        && testParams[1].getType().equals(SearchClientAdapter.class));
+  }
+
+  @Override
+  public Stream<TestTemplateInvocationContext> provideTestTemplateInvocationContexts(
+      final ExtensionContext extensionContext) {
+    final var osConfig = getConfigWithConnectionDetails(OPENSEARCH);
+    final var osClient = new OpensearchConnector(osConfig.getConnect()).createClient();
+    final var osClientAdapter = new SearchClientAdapter(osClient);
+
+    final var elsConfig = getConfigWithConnectionDetails(ELASTICSEARCH);
+    final var elsClient = new ElasticsearchConnector(elsConfig.getConnect()).createClient();
+    final var elsClientAdapter = new SearchClientAdapter(elsClient);
+
+    try {
+      elsClient.indices().delete(req -> req.index("*"));
+      elsClient.indices().deleteIndexTemplate(req -> req.name("*"));
+      osClient.indices().delete(req -> req.index("*"));
+      osClient.indices().deleteIndexTemplate(req -> req.name("*"));
+    } catch (final Exception e) {
+      throw new RuntimeException("Failed to reset container data", e);
+    }
```

Comment:
🔧 same as in previous PR, is there a way to only reset data for the client under test not both? if there's a simple way to check and do it, it would be great.

Reply:
The clean up here occurs before providing the invocation context to a test template, so it works as you want e.g.
1. clean up ols and es
2. run els test case
3. run os test case
4. next test template 
5. clean up ols and es
6. etc